packages/create-ima-app/template/app/page/home/__tests__/HomeViewIntegration.js

```diff
@@ -0,0 +1,45 @@
+import { initImaApp, clearImaApp } from '@ima/plugin-testing-integration';
+import cards from '../../../public/cards.json';
+
+describe('Home page', () => {
+  const response = {
+    body: cards,
+  };
+  let app;
+  let http;
+
+  beforeEach(async () => {
+    app = await initImaApp({
+      initBindApp: (ns, oc) => {
+        http = oc.get('$Http');
+
+        // Mock http.get method so the application
+        // wont make any external requests
+        // and return mocked response
+        http.get = jest.fn().mockReturnValue(Promise.resolve(response));
+      },
+    });
+
+    await app.oc.get('$Router').route('/');
+  });
+
+  afterEach(() => {
+    clearImaApp(app);
+  });
+
+  it('can render component', async () => {
+    expect(document.querySelectorAll('.cards')).toHaveLength(1);
+    expect(document.querySelectorAll('.card')).toHaveLength(cards.length);
+
+    for (let i = 0; i < cards.length; i++) {
+      let paragraphText = document
```

Comment:
We are calling here (and few line later again) `document.querySelectorAll` repeatedly with same argument and select specific element from the array. Why don't we call these query selectors only once and then in the cycle we can just select specific index from the array?

Reply:
Query selectors call only once before cycle.